src/sentry/south_migrations/0463_backfill_service_hook_project.py

```diff
@@ -0,0 +1,1344 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+
+    # Flag to indicate if this migration is too risky
+    # to run online and needs to be coordinated for offline
+    is_dangerous = False
+
+    def forwards(self, orm):
+        db.commit_transaction()
+        try:
+            self._forwards(orm)
+        except Exception:
+            db.start_transaction()
+            raise
+        db.start_transaction()
+
+    def _forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Don't use "from appname.models import ModelName".
+        # Use orm.ModelName to refer to models in this application,
+        # and orm['appname.ModelName'] for models in other applications.
+        ServiceHook = orm['sentry.ServiceHook']
+        ServiceHookProject = orm['sentry.ServiceHookProject']
+        Project = orm['sentry.Project']
+
+        queryset = ServiceHook.objects.all()
+
+        for service_hook in RangeQuerySetWrapperWithProgressBar(queryset):
+            try:
+                ServiceHookProject.objects.create(
+                    project_id=service_hook.project_id,
+                    service_hook_id=service_hook.id,
+                )
+            except IntegrityError:
```

Comment:
You need to wrap all of this in a `transaction.atomic()` though or this will be problematic.